kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/CommitterImpl.java

```diff
@@ -0,0 +1,253 @@
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.util.Pair;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ListConsumerGroupOffsetsOptions;
+import org.apache.kafka.clients.admin.ListConsumerGroupOffsetsResult;
+import org.apache.kafka.clients.consumer.ConsumerGroupMetadata;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.clients.producer.Producer;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.sink.SinkTaskContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CommitterImpl implements Committer, Closeable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CommitterImpl.class);
+  private final SinkTaskContext context;
+  private final IcebergSinkConfig config;
+  private final boolean isLeader;
+  private final CoordinatorThread coordinatorThread;
+  private final CommitRequestListener commitRequestListener;
+  private final UUID producerId;
+  private final Producer<String, byte[]> producer;
+
+  public CommitterImpl(SinkTaskContext context, IcebergSinkConfig config) {
+    this(
+        context,
+        config,
+        new ConsumerFactoryImpl(),
+        new AdminFactoryImpl(),
+        new ProducerFactoryImpl());
+  }
+
+  private CommitterImpl(
+      SinkTaskContext context,
+      IcebergSinkConfig config,
+      ConsumerFactory consumerFactory,
+      AdminFactory adminFactory,
+      ProducerFactory producerFactory) {
+    this(
+        context,
+        config,
+        new ControlTopicCommitRequestListener(config, consumerFactory),
+        adminFactory,
+        producerFactory,
+        new CoordinatorThreadFactoryImpl(adminFactory, consumerFactory, producerFactory));
+  }
+
+  @VisibleForTesting
+  CommitterImpl(
+      SinkTaskContext context,
+      IcebergSinkConfig config,
+      CommitRequestListener commitRequestListener,
+      AdminFactory adminFactory,
+      ProducerFactory producerFactory,
+      CoordinatorThreadFactory coordinatorThreadFactory) {
+    this.context = context;
+    this.config = config;
+    this.isLeader = Objects.equals(config.taskId(), 0);
+
+    if (isLeader) {
+      this.coordinatorThread = coordinatorThreadFactory.create(config);
+    } else {
+      this.coordinatorThread = null;
+    }
+
+    this.commitRequestListener = commitRequestListener;
+
+    Map<String, String> producerProps = Maps.newHashMap(config.kafkaProps());
+    // use a random transactional-id to avoid producer generation based fencing
+    producerProps.put(
+        ProducerConfig.TRANSACTIONAL_ID_CONFIG,
+        String.join(
+            "-",
+            config.connectGroupId(),
+            config.taskId().toString(),
+            "committer",
+            UUID.randomUUID().toString()));
```

Comment:
Another minor change (i.e. I do not consider this a breaking change): using a more read-able/identifiable transactional-id.
This would be useful in situations if you ever have a hanging transaction that you want to abort manually outside of kafka connect.